### PR TITLE
[refactor] Add serializer_spec

### LINKED ISSUE #43: Delegate array methods for Relation
Currently, we have to do where.to_a.first, which is a little too verbose.

### PR DESCRIPTION
**Pull request by donaldong**
This adds serializer_spec, fixes some rubocop lint errors, and delegates all the related array methods to execute queries.

### CODE REVIEW

File: lib/redcord/attribute.rb [hunk 1]
```diff
@@ -58,6 +58,11 @@ def ttl(duration)
       class_variable_set(:@@ttl, duration)
     end
 
+    sig { returns(String) }
```
Comment (katyho) [inline lib/redcord/attribute.rb:61]: Why do you need to define this here? This is also defined in serializer

File: lib/redcord/relation.rb [hunk 2]
```diff
@@ -67,24 +66,93 @@ def count
     redis.find_by_attr_count(model.model_key, query_conditions)
   end
 
-  sig { returns(T::Array[T.untyped]) }
-  def to_a
-    execute_query
-  end
+  delegate(
```
Comment (katyho) [inline lib/redcord/relation.rb:69]: Just wondering how you came up with this extensive list?

File: spec/serializer_spec.rb [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    another_instance = klass.find_by(id: instance.id)
+
+    expect(instance.d).to be_nil
+    expect(instance.created_at.to_i).to eq(now.to_i)
+    expect(instance.updated_at.to_i).to eq(now.to_i)
+    expect(instance.created_at.to_i).to eq(another_instance.created_at.to_i)
+    expect(instance.updated_at.to_i).to eq(another_instance.updated_at.to_i)
+  end
+
+  it 'works with boolean types' do
```
Comment (katyho) [inline spec/serializer_spec.rb:42]: What is the value of this test specifically? Is it just to check that a boolean value is serialized/deserialized correctly by the redis client?

File: spec/serializer_spec.rb [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+    queried_instances = klass.where(
+      d: Redcord::RangeInterval.new(max: Time.zone.now + 1.day),
+    )
+    expect(queried_instances.size).to eq 2
+    expect(queried_instances.first.id).to eq(first.id).or eq(second.id)
+    expect(queried_instances.second.id).to eq(first.id).or eq(second.id)
+  end
+
+  it 'allows time types to be range index attributes' do
```
Comment (katyho) [inline spec/serializer_spec.rb:135]: Is this a duplicate of the test above?

File: lib/redcord/attribute.rb (hunk 1 above)
Comment (donaldong) [reply lib/redcord/attribute.rb:61]: I moved this from serializer.rb to attribute.rb 🤔 I guess serializer.rb  is a better place for this method after all. Will put it back lol

File: lib/redcord/relation.rb (hunk 2 above)
Comment (donaldong) [reply lib/redcord/relation.rb:69]: I went through https://ruby-doc.org/core-2.7.0/Array.html#method-i-7C and inspected existing methods on an array (with active support / array loaded). I selected most related methods to delegate.

File: spec/serializer_spec.rb (hunk 3 above)
Comment (donaldong) [reply spec/serializer_spec.rb:42]: yep!

File: spec/serializer_spec.rb (hunk 4 above)
Comment (donaldong) [reply spec/serializer_spec.rb:135]: huh 🤔 idk how that happened. Good catch!

**Comment by donaldong** [comment]
address comments

:mag_right: [Click to see changes since last update](https://github.com/chanzuckerberg/redcord/compare/fcd2617ddf302275f540eba663cbbc2e07cddb7c..bfa44d23a95cbc678e75135b8e0c79e5e62b88d6).

**Review by katyho — APPROVED** [review]

**Pull request merged**